### ISSUE TITLE
Integration of Android Studio 4.1 with Youtrack is broken

### ISSUE THREAD

**Issue by dmitry-fbm** [jk1/youtrack-idea-plugin#143]
In AS 4.1 integration with JetBrains Youtrack has changed from username/password to Youtrack token. When I input the token in configuration popup and push test button it reports successful connection. However nothing appears in Youtrack toolbar, it still says not configured; after pushing refresh button status bar message says: cannot connect to server, are you offline? This lasts for a while, and after some (long) time with no action on my side it magically connects and starts to receive updates. Am not 100% sure but so far my impression was that it connects after someone else made an update in Youtrack that has to be delivered to me.
Another issue is that after restarting AS the token gets lost and I have to input it again every time. This actually isn't quite new problem and perhaps isn't related to the plugin itself - I quite often (not every time, but in most cases) had the same trouble yet in AS 4.0 using old user/password authentication method, and experience the same behavior with GitLab integration plugin.

That's very annoying, please fix ASAP.

Thanks

Windows 10 2004 Build 19041.508
Android Studio 4.1
Build #AI-201.8743.12.41.6858069, built on September 23, 2020
Runtime version: 1.8.0_242-release-1644-b01 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Windows 10 10.0
GC: ParNew, ConcurrentMarkSweep
Memory: 1237M
Cores: 8
Registry: ide.new.welcome.screen.force=true, ide.tree.collapse.recursively=false
Non-Bundled Plugins: Batch Scripts Support, CMD Support, color.scheme.GapStyle, String Manipulation, com.herbert.george.flutter-snippets, com.github.jk1.ytplugin, org.intellij.plugins.markdown, org.jetbrains.kotlin, com.rhyme.flutter.plugin.jumptoassets, mobi.hsz.idea.gitignore, net.seesharpsoft.intellij.plugins.file-preview, Dart, izhangzhihao.rainbow.brackets, io.flutter, de.mariushoefler.flutter_enhancement_suite, com.chenenyu.intellij.widgetgenerator, com.localizely.flutter-intl

YouTrack Build 2020.3.11807 Mon, Sep 28, 2020, 12:22:41 PM UTC

**Comment by jk1** [comment]
Hi,

First of all, let's check if your IDE is allowed to remember passwords. To do so navigate to File | Settings | Appearance & Behavior | System Settings | Passwords and make sure that password persistence is permitted.

**Comment by dmitry-fbm** [comment]
Hello, thanks for your advise - never knew such setting exists and for some reason it was set to "do not remember". Ok it took several input/restart iterations for IDE to remember passwords but in the end it worked and solved also the original issue of being unable to connect to YT. Still you may wish to solve connection issue in case of "do not remember" passwords saving policy.

Thanks, my problems are solved so it's up to you to close the ticket or keep it open.

**Comment by jk1** [comment]
@dmitry-fbm thank you for the update
@lesslyrics let's make sure we're able to establish the connection even with "do not remember" policy enabled. In addition to that, it wouldn't hurt to display a warning near the token field if the policy is effective.

**Comment by dmitry-fbm** [comment]
Ok there is another scenario you may wish to check: 
- start AS connected to YT, make sure that connection is established;
- hibernate the computer
- restore computer from hibernated state
- push refresh on YT toolbar
In my case at least it doesn't connect, displaying the same "are you offline?". Restarting AS solves this.

**Comment by cornelRaiu** [comment]
I am receiving the message as well in PHPStorm:

![image](https://user-images.githubusercontent.com/6166479/98052730-9cf4e480-1e3f-11eb-9bab-cd567baee1af.png)

Everything worked perfectly for about 2 weeks until about 1 hour ago.

My settings are set to save passwords in the KeePass. I tried to uninstall the YouTrack Integration plugin, I regenerated the token, all to no avail.

The weird part is that I could still update the status and add spent time to the issues that I already had in my list but I could not add comments or update the issues list whatever I did.

Then, I changed my login settings to password and it works correctly again.

However, this is a pretty annoying bug.
